### PR TITLE
Update needle definitions to v1.4.3

### PR DESCRIPTION
**Pull request by nikeee**
This PR updates the definitions for needle to match the docs/source of v1.4.3. Since this is a new major version, the old definition was moved to the v0 folder.
Docs + Source: https://github.com/tomas/needle
npm package: https://www.npmjs.com/package/needle

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/tomas/needle
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

File: needle/index.d.ts [hunk 1]
```diff
@@ -1,92 +1,118 @@
...
-            put(url: string, data: any, callback?: Callback): ReadableStream;
-            put(url: string, data: any, options?: RequestOptions, callback?: Callback): ReadableStream;
+            patch(url: string, data: BodyData, callback?: NeedleCallback): ReadableStream;
+            patch(url: string, data: BodyData, options?: NeedleOptions, callback?: NeedleCallback): ReadableStream;
 
-            delete(url: string, data: any): ReadableStream;
-            delete(url: string, data: any, callback?: Callback): ReadableStream;
-            delete(url: string, data: any, options?: RequestOptions, callback?: Callback): ReadableStream;
+            delete(url: string, data: BodyData, callback ?: NeedleCallback): ReadableStream;
+            delete(url: string, data: BodyData, options ?: NeedleOptions, callback ?: NeedleCallback): ReadableStream;
```
Comment (DanielRosenwasser) [inline needle/index.d.ts]: can you remove the space between the parameter name and the `?`

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

**Comment by DanielRosenwasser** [comment]
@bigsan can you take a quick look/

File: needle/index.d.ts (hunk 1 above)
Comment (nikeee) [inline needle/index.d.ts]: Should be fixed in 16e6ab0

**Comment by bowdenk7** [comment]
No response from other authors for over 5 days. Merging.

**Pull request merged**